ambry-server/src/integration-test/java/com.github.ambry.server/RouterServerPlaintextTest.java

```diff
@@ -0,0 +1,140 @@
+/**
+ * Copyright 2016 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ */
+package com.github.ambry.server;
+
+import com.github.ambry.server.RouterServerTestFramework.OperationInfo;
+import com.github.ambry.server.RouterServerTestFramework.OperationType;
+import com.github.ambry.utils.SystemTime;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Queue;
+import org.apache.log4j.Level;
+import org.apache.log4j.Logger;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import static com.github.ambry.server.RouterServerTestFramework.getRouterProperties;
+
+
+public class RouterServerPlaintextTest {
+  private static MockCluster plaintextCluster;
+  private static RouterServerTestFramework testFramework;
+
+  @BeforeClass
+  public static void initializeTests()
+      throws Exception {
+//    Logger.getRootLogger().setLevel(Level.ERROR);
+//    org.apache.log4j.BasicConfigurator.configure();
+
+    MockNotificationSystem notificationSystem = new MockNotificationSystem(9);
+    plaintextCluster = new MockCluster(notificationSystem, false, SystemTime.getInstance());
+    plaintextCluster.startServers();
+    testFramework = new RouterServerTestFramework(getRouterProperties("DC1"), plaintextCluster, notificationSystem);
+  }
+
+  @AfterClass
+  public static void cleanup()
+      throws IOException {
+    testFramework.cleanup();
+    long start = System.currentTimeMillis();
+    System.out.println("About to invoke cluster.cleanup()");
+    if (plaintextCluster != null) {
+      plaintextCluster.cleanup();
+    }
+    System.out.println("cluster.cleanup() took " + (System.currentTimeMillis() - start) + " ms.");
+  }
+
+  @Test
+  public void interleavedOperationsTest()
+      throws Exception {
+    List<OperationInfo> opInfos = new ArrayList<>();
+    for (int i = 0; i < 100; i++) {
+      Queue<OperationType> opChain = new LinkedList<>();
```

Comment:
This seems a pretty random operation sequence. Instead of looping for 100 times, my suggestion is to try some different sequences for completeness. I did some small test e.g., inserting one more `opChain.add(OperationType.GET_INFO_NB);` after line 71 it still passes, but this forms a different operation sequence and increases the coverage for test cases.